component/src/main/java/io/siddhi/extension/store/mongodb/MongoSelectExpressionVisitor.java

```diff
@@ -0,0 +1,307 @@
+                conditionOperands.push("{$" + functionName + ":" + functionArgument + "}");
+            }
+        }
+    }
+
+    @Override
+    public void beginVisitParameterAttributeFunction(int index) {
+    }
+
+    @Override
+    public void endVisitParameterAttributeFunction(int index) {
+    }
+
+    @Override
+    public void beginVisitMath(MathOperator mathOperator) {
+    }
+
+    @Override
+    public void endVisitMath(MathOperator mathOperator) {
+        String rightOperand = this.conditionOperands.pop();
+        String leftOperand = this.conditionOperands.pop();
+        conditionOperands.push("{$" + mathOperator.name().toLowerCase() + ":" + "[" + leftOperand + "," +
+                rightOperand + "]}");
+    }
+
+    @Override
+    public void beginVisitMathLeftOperand(MathOperator mathOperator) {
+    }
+
+    @Override
+    public void endVisitMathLeftOperand(MathOperator mathOperator) {
+    }
+
+    @Override
+    public void beginVisitMathRightOperand(MathOperator mathOperator) {
+    }
+
+    @Override
+    public void endVisitMathRightOperand(MathOperator mathOperator) {
+    }
+
+    @Override
+    public void beginVisitOr() {
+    }
+
+    @Override
+    public void endVisitOr() {
+        String rightOperand = this.conditionOperands.pop();
+        String leftOperand = this.conditionOperands.pop();
+        conditionOperands.push("{$or:" + "[" + leftOperand + "," + rightOperand + "]}");
+    }
+
+    @Override
+    public void beginVisitOrLeftOperand() {
+    }
+
+    @Override
+    public void endVisitOrLeftOperand() {
+    }
+
+    @Override
+    public void beginVisitOrRightOperand() {
+    }
+
+    @Override
+    public void endVisitOrRightOperand() {
+    }
+
+    @Override
+    public void beginVisitAnd() {
+    }
+
+    @Override
+    public void endVisitAnd() {
+        String rightOperand = this.conditionOperands.pop();
+        String leftOperand = this.conditionOperands.pop();
+        conditionOperands.push("{$and:" + "[" + leftOperand + "," + rightOperand + "]}");
+    }
+
+    @Override
+    public void beginVisitAndLeftOperand() {
+    }
+
+    @Override
+    public void endVisitAndLeftOperand() {
+    }
+
+    @Override
+    public void beginVisitAndRightOperand() {
+    }
+
+    @Override
+    public void endVisitAndRightOperand() {
+    }
+
+    @Override
+    public void beginVisitIsNull(String streamId) {
+        this.isNullCheck = true;
+    }
+
+    @Override
+    public void endVisitIsNull(String streamId) {
+        this.isNullCheck = false;
+        String nullCheckAttribute = conditionOperands.pop();
+        conditionOperands.push("{ $cond: { if: { $eq: [" + nullCheckAttribute + ", null ] }, then: true, " +
+                "else: false } }");
+    }
+
+    @Override
+    public void beginVisitCompare(Compare.Operator operator) {
+    }
+
+    @Override
+    public void endVisitCompare(Compare.Operator operator) {
+        String compareFilter;
+        switch (operator) {
```

Comment:
The checks and logic seem to be same as in MongoExpressionVisitor.endVisitCompare() except the wrapping of `{ $cond: { if: { }}}` see if you can reuse the method by moving them to util class.

Reply:
Fixed in
https://github.com/siddhi-io/siddhi-store-mongodb/pull/56/commits/19316e5642550ec5d4d154a11df69b7c14d8d726